### PR TITLE
fix: correctly initialize the copy properties completion time of a blob properties

### PR DESCRIPTION
**Pull request by hector-sab**
# Description

Corrects initialization of the copy properties completion time attribute. In its current form, the completion time is set to None as the key is using the incorrect keyname.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/storage/azure-storage-blob/azure/storage/blob/_models.py [hunk 1]
```diff
@@ -651,7 +651,7 @@ def __init__(self, **kwargs):
         self.source = kwargs.get('x-ms-copy-source')
         self.status = get_enum_value(kwargs.get('x-ms-copy-status'))
         self.progress = kwargs.get('x-ms-copy-progress')
-        self.completion_time = kwargs.get('x-ms-copy-completion_time')
+        self.completion_time = kwargs.get('x-ms-copy-completion-time')
```
Comment (vincenttran-msft) [inline sdk/storage/azure-storage-blob/azure/storage/blob/_models.py:654]: This is a great fix. Thanks! If you could add this to the `blobs` package CHANGELOG under `Bugs Fixed`
"- Fixed a typo in the initialization of `completion_time` for the `CopyProperties` model."

With that addition, we will likely be able to take this pull-request as is. Thanks for your community contribution!

**Review by vincenttran-msft — CHANGES_REQUESTED** [review]
Add changelog entry:
" - Fixed a typo in the initialization of `completion_time` for the `CopyProperties` model. "

File: sdk/storage/azure-storage-blob/azure/storage/blob/_models.py (hunk 1 above)
Comment (hector-sab) [reply sdk/storage/azure-storage-blob/azure/storage/blob/_models.py:654]: @vincenttran-msft Just added the entry. Let me know if that works :)

**Comment by hector-sab** [comment]
@microsoft-github-policy-service agree

File: sdk/storage/azure-storage-blob/CHANGELOG.md [hunk 2]
```diff
@@ -13,6 +13,7 @@ tokens to be used with multiple services. By default, the SAS token service scop
 Python 3.12.
 - Fixed an issue where authentication errors could raise `AttributeError` instead of `ClientAuthenticationError` when
 using async OAuth credentials.
+ - Fixed a typo in the initialization of `completion_time` for the `CopyProperties` model.
```
Comment (vincenttran-msft) [inline sdk/storage/azure-storage-blob/CHANGELOG.md:16]: Nitpick: There is a leading whitespace on this line. Will be perfect after that fix 😄

File: sdk/storage/azure-storage-blob/azure/storage/blob/_models.py (hunk 1 above)
Comment (vincenttran-msft) [reply sdk/storage/azure-storage-blob/azure/storage/blob/_models.py:654]: Thanks, left a response. Also I may have caused a merge conflict in the changelog (merged a different PR with a changelog entry) so you may need to resolve that!

**Review by vincenttran-msft — CHANGES_REQUESTED** [review]

File: sdk/storage/azure-storage-blob/CHANGELOG.md (hunk 2 above)
Comment (hector-sab) [reply sdk/storage/azure-storage-blob/CHANGELOG.md:16]: Shoot... I'll fix it later today. Amazing catch!

**Comment by hector-sab** [comment]
Hey @vincenttran-msft . So I made this PR worst... I'm going to close this one instead and just move the PR to this new one.  https://github.com/Azure/azure-sdk-for-python/pull/34559

Cheers

**Pull request closed without merging**